src/agent.ts

```diff
@@ -211,7 +215,7 @@ export class Agent {
         messages: this._chat.messages,
         functions: functionDescriptions,
         temperature: 0,
-        max_tokens: 500
+        max_tokens: Number(env().OPENAI_API_KEY)
```

Comment:
This is wrong, we shouldn't be using "OPENAI_API_KEY" here. The fact that you have to cast the type should tell you something is wrong.